GitUI/HelperDialogs/FormResetAnotherBranch.cs

```diff
@@ -93,7 +94,7 @@ private void Ok_Click(object sender, EventArgs e)
                 return;
             }
 
-            var command = GitCommandHelpers.PushLocalCmd(gitRefToReset.CompleteName, _revision.ObjectId, Module.WorkingDir, ForceReset.Checked);
+            var command = GitCommandHelpers.PushLocalCmd(gitRefToReset.CompleteName, _revision.ObjectId, Module.GetRepoPath(Module.WorkingDir), ForceReset.Checked);
```

Comment:
We need to clearly define and explain the rules when `GetRepoPath` must be used.

Also any reason not to compute and cache this property when we initialise a `GitModule`?

Reply:
GetRepoPath is required for paths that the Git executable will use.
Maybe rename to GetPathForGitExe()?

Reply:
> GetRepoPath is required for paths that the Git executable will use.

Should this be pushed from UI layer into the GitCommands layer then? That is, instead of forcing callers to remember/know that paths must be encoded in a specific way, should callee (i.e. git commands, git executable, etc.) perform encoding when appropriate?

Reply:
What I'm getting at - is "use WSL" something a consumer should care, or is it an infra setting that GitModule and Co should take care of based on the characteristics of a repo?

Reply:
This handles specific arguments only. 
An alternative would be to guess what are file arguments to transform when running the Git command (ths is what the wslgit wrapper does). The problem with that is that it is hard to find what is arguments. You may translate for instance searches for "C:\repo" and it is hard to handle where paths end if the paths contain spaces.

For the global "use WSL" configuration, the user will need to know about this as some settings will have to be done separately in WSL. But there should be no need to know details.

Reply:
Renamed the method, hopefully clearer now